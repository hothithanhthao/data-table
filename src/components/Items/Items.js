import React, { useState } from 'react';
import { formatDate } from '../common';
import './Items.css'

const sortTypes = {
  up: {
    fn: (a, b) => a.timestamp - b.timestamp
  },
  down: {
    fn: (a, b) => b.timestamp - a.timestamp
  },
  default: {
    fn: (a, b) => b.timestamp - a.timestamp
  }
}


export const Items = ({rows}) => {
  const [currentSort, setCurrentSort] = useState('default');

  const onSortChange = () => {
		let nextSort;

		if (currentSort === 'down') nextSort = 'up';
		else if (currentSort === 'up') nextSort = 'default';
		else if (currentSort === 'default') nextSort = 'down';

		setCurrentSort(nextSort);
	};

  return (
    rows.length > 0 && (
      <table className='text-left'>
        <thead>
          <tr>
            <th>
              <button onClick={onSortChange}>Date</button>
            </th>
            <th>User ID</th>
            <th>Old value</th>
            <th>New value</th>
          </tr>
        </thead>
        <tbody>
          {[...rows].sort(sortTypes[currentSort].fn).map((item, index) => (
            <tr key={index}>
              <td>{formatDate(item.timestamp)}</td>
              <td>{item.id}</td>
              <td>{item.diff[0].oldValue}</td>
            <td>{item.diff[0].newValue}</td>
            </tr>
          ))}
        </tbody>
      </table>
    )
  );
};

export default Items;
