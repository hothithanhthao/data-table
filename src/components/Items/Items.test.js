import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {Items} from './Items';

const rows = [
  {
    id: 'e28d290a-a2f2-48c2-9001-ff43884e271b',
    timestamp: new Date('2020/2/14').getTime(),
    diff: [
      { field: 'name', oldValue: 'John', newValue: 'Bruce' },
    ],
  },
  {
    id: '8bd0166f-f0c6-48fd-9fcd-a17e76eb1e92',
    timestamp: new Date('2020/2/15').getTime(),
    diff: [
      { field: 'name', oldValue: 'Bruce', newValue: 'Nick' },
    ],
  }];

describe('Items', () => {
  it('renders according to snapshot', () => {
    const sallowComponent = shallow(
      <Items rows={rows}/>
    );
    expect(toJson(sallowComponent)).toMatchSnapshot();
 });
});
