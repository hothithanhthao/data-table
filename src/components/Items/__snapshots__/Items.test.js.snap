// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items renders according to snapshot 1`] = `
<table
  className="text-left"
>
  <thead>
    <tr>
      <th>
        <button
          onClick={[Function]}
        >
          Date
        </button>
      </th>
      <th>
        User ID
      </th>
      <th>
        Old value
      </th>
      <th>
        New value
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        2020-02-15
      </td>
      <td>
        8bd0166f-f0c6-48fd-9fcd-a17e76eb1e92
      </td>
      <td>
        Bruce
      </td>
      <td>
        Nick
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        2020-02-14
      </td>
      <td>
        e28d290a-a2f2-48c2-9001-ff43884e271b
      </td>
      <td>
        John
      </td>
      <td>
        Bruce
      </td>
    </tr>
  </tbody>
</table>
`;
